servicetalk-concurrent-test-internal/src/main/java/io/servicetalk/concurrent/test/internal/AwaitUtils.java

```diff
@@ -167,14 +167,36 @@ public static boolean awaitUninterruptibly(CountDownLatch latch, long timeout, T
         }
     }
 
+    /**
+     * {@link BlockingQueue#poll(long, TimeUnit)} from the queue, throws unchecked Exception if
+     * thread is interrupted while waiting.
+     * @param queue the queue to poll from.
+     * @param timeout the timeout duration to poll for.
+     * @param unit the units applied to {@code timeout}.
+     * @param <T> The types of objects in the queue.
+     * @return see {@link BlockingQueue#poll(long, TimeUnit)}.
+     */
+    @Nullable
+    public static <T> T poll(BlockingQueue<T> queue, long timeout, TimeUnit unit) {
+        try {
+            return queue.poll(timeout, NANOSECONDS);
```

Comment:
The `unit` is unused and `NANOSECONDS` is enforced regardless of the intention of the caller. I believe that's unintended.

Reply:
@chemicL Good catch, fixed it, thanks. `pollUninterruptibly` uses `NANOSECONDS` by default, is that because the timeout is manually controlled by the `waitTime`? Here is the code of the `pollUninterruptibly`.
```
public static <T> T pollUninterruptibly(BlockingQueue<T> queue, long timeout, TimeUnit unit) {
        final long startTime = System.nanoTime();
        final long timeoutNanos = NANOSECONDS.convert(timeout, unit);
        long waitTime = timeout;
        boolean interrupted = false;
        try {
            do {
                try {
                    return queue.poll(waitTime, NANOSECONDS);
                } catch (InterruptedException e) {
                    interrupted = true;
                }
                waitTime = timeoutNanos - (System.nanoTime() - startTime);
                if (waitTime <= 0) {
                    return null;
                }
            } while (true);
        } finally {
            if (interrupted) {
                Thread.currentThread().interrupt();
            }
        }
    }

Reply:
Apologies for missing the notification. I think the only reason there is the `System.nanoTime()` call, which provides nanos, so the calculations are based around that unit. Any JDK method accepting a `long` with the specifier for the `TimeUnit` that the value represents will deal fine with any unit. Please note that the `BlockingQueue#poll(long, TimeUnit)` method internally transforms the values to nanos, so passing nanos here is the most straightforward and doesn't require any additional computation. Hope that helps.